cmd/backup-manager/app/export/manager.go

```diff
@@ -323,6 +323,23 @@ func (bm *BackupManager) performBackup(backup *v1alpha1.Backup, db *sql.DB) erro
 
 	remotePath := strings.TrimPrefix(archiveBackupPath, constants.BackupRootPath+"/")
 	bucketURI := bm.getDestBucketURI(remotePath)
+	backup.Status.BackupPath = bucketURI
+	err = bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
+		Type:   v1alpha1.BackupPrepare,
+		Status: corev1.ConditionTrue,
+	})
+	if err != nil {
+		errs = append(errs, err)
+		uerr := bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
+			Type:    v1alpha1.BackupFailed,
+			Status:  corev1.ConditionTrue,
+			Reason:  "UpdateBackupPathFailed",
+			Message: err.Error(),
+		})
```

Comment:
why setting this condition if `StatusUpdater.Update(` fails?

Reply:
copyed from https://github.com/pingcap/tidb-operator/blob/master/cmd/backup-manager/app/backup/manager.go#L159
It sames that it's better to return error directly?

Reply:
IMO it's meaningless to call `StatusUpdater.Update` again if the previous `StatusUpdater.Update` fails as the second call can fail too

our current architecture relies on our backup job to update `Backup` CR's status, the situation that the status may not be reported to the apiserver is unavoidable

does it cause problems when the `StatusUpdater.Update` call fails?

Reply:
> copyed from https://github.com/pingcap/tidb-operator/blob/master/cmd/backup-manager/app/backup/manager.go#L159
> It sames that it's better to return error directly?

yes

Reply:
If we omit this error, we might be unable to clean files on the Cloud when we delete this CR because `BackupPath` is not recorded.

Reply:
but the `bm.backupDataToRemote` will not be run if reporting the backup path to the apiserver fails

(if this is not true, I think we should fix it. one more API call does not help)

Reply:
So should we omit this error? What about BR?

Reply:
we didn't omit the error, the job fails